### PR TITLE
Turbopack: Don’t show stack traces for internal modules

### PR DESCRIPTION
**Pull request by wbinnssmith**
See https://github.com/vercel/next.js/blob/ac7607f9775ea97bcb7f01f9da148d8c51abcc42/packages/next/src/client/components/react-dev-overlay/server/shared.ts#L59-L62.

Test Plan: Added an e2e test.

Closes PACK-2919

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: a119dbe1555778fc5dd259943bcf977ff166cff0

`TURBOPACK=1 pnpm test-start test/e2e/internal-traces/internal-traces.test.ts` (turbopack)
- internal traces > should not write long internal traces to stdio

<details>
<summary>Expand output</summary>

  ● internal traces › should not write long internal traces to stdio

    next build failed with code/signal 1

      89 |           if (code || signal)
      90 |             reject(
    > 91 |               new Error(`next build failed with code/signal ${code || signal}`)
         |               ^
      92 |             )
      93 |           else resolve()
      94 |         })

      at ChildProcess.<anonymous> (lib/next-modes/next-start.ts:91:15)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| buildDuration | 14s | 13.9s | N/A |
| buildDurationCached | 7.5s | 6.2s | N/A |
| nodeModulesSize | 199 MB | 199 MB | ⚠️ +5.32 kB |
| nextStartRea..uration (ms) | 395ms | 407ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 31.4 kB | 31.4 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | ✓ |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 242 B | 241 B | N/A |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 99 kB | 99 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| index.html gzip | 529 B | 528 B | N/A |
| link.html gzip | 541 B | 541 B | ✓ |
| withRouter.html gzip | 524 B | 522 B | N/A |
| Overall change | 541 B | 541 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.4 kB | 95.4 kB | N/A |
| page.js gzip | 3.06 kB | 3.06 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 626 B | 626 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 97 kB | 97 kB | ✓ |
| app-page-tur..prod.js gzip | 98.8 kB | 98.8 kB | ✓ |
| app-page-tur..prod.js gzip | 93 kB | 93 kB | ✓ |
| app-page.run...dev.js gzip | 144 kB | 144 kB | ✓ |
| app-page.run..prod.js gzip | 91.5 kB | 91.5 kB | ✓ |
| app-route-ex...dev.js gzip | 21.4 kB | 21.4 kB | ✓ |
| app-route-ex..prod.js gzip | 15.1 kB | 15.1 kB | ✓ |
| app-route-tu..prod.js gzip | 15.1 kB | 15.1 kB | ✓ |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route.ru...dev.js gzip | 21.1 kB | 21.1 kB | ✓ |
| app-route.ru..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-turbo...prod.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim...dev.js gzip | 23.1 kB | 23.1 kB | ✓ |
| pages.runtim..prod.js gzip | 22.4 kB | 22.4 kB | ✓ |
| server.runti..prod.js gzip | 51.1 kB | 51.1 kB | ✓ |
| Overall change | 945 kB | 945 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong></summary>

|  | vercel/next.js canary  | vercel/next.js wbinnssmith/internal-traces | Change |
| - | - | - | - |
| 0.pack gzip | 1.58 MB | 1.58 MB | N/A |
| index.pack gzip | 106 kB | 106 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: a119dbe1555778fc5dd259943bcf977ff166cff0

File: test/development/acceptance/ReactRefreshLogBox-builtins.test.ts [hunk 1]
```diff
@@ -49,9 +49,6 @@ describe.each(['default', 'turbo'])('ReactRefreshLogBox %s', () => {
       expect(await session.getRedboxSource()).toMatchInlineSnapshot(`
         "./node_modules/my-package/index.js:1:13
         Module not found: Can't resolve 'dns'
-        > 1 | const dns = require('dns')
```
Comment (timneutkens) [inline test/development/acceptance/ReactRefreshLogBox-builtins.test.ts:52]: For node_modules the codeframe should still be shown, just not for ones that are very few lines and the individual line is thousands of characters.

Comment (wbinnssmith) [reply test/development/acceptance/ReactRefreshLogBox-builtins.test.ts:52]: This is consistent with webpack's behavior — should we change both?

File: test/development/acceptance/ReactRefreshLogBox-builtins.test.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ describe.each(['default', 'turbo'])('ReactRefreshLogBox %s', () => {
   })
 
   // Module trace is only available with webpack 5
-  test('Node.js builtins', async () => {
+  test.only('Node.js builtins', async () => {
```
Comment (kwonoj) [inline test/development/acceptance/ReactRefreshLogBox-builtins.test.ts:14]: is this expected?

File: test/development/acceptance/ReactRefreshLogBox-builtins.test.ts [hunk 3]
```diff
@@ -59,6 +59,9 @@ describe.each(['default', 'turbo'])('ReactRefreshLogBox %s', () => {
       expect(await session.getRedboxSource()).toMatchInlineSnapshot(`
         "./node_modules/my-package/index.js:1:1
         Module not found: Can't resolve 'dns'
+        > 1 | const dns = require('dns')
```
Comment (kwonoj) [inline test/development/acceptance/ReactRefreshLogBox-builtins.test.ts:62]: what's difference between https://github.com/vercel/next.js/pull/64228/files#diff-05b6ff60dd6730bcd70eb57fc3313401ed96c5e427468d2aa2758894541e2c98L57 one removes existing stack and one adds new?

Comment (wbinnssmith) [reply test/development/acceptance/ReactRefreshLogBox-builtins.test.ts:62]: This brings Next.js+Webpack in line with the Turbopack behavior, showing these frames (but not long bundled internal files)

**Review by bgw — APPROVED** [review]

**Pull request merged**